community/logging/src/main/java/org/neo4j/logging/RotatingFileOutputStreamSupplier.java

```diff
@@ -292,26 +294,34 @@ private void rotate()
                         if ( !closed.get() && outRef.equals( nullStream ) )
                         {
                             outRef = openOutputFile();
-                            rotationListener.outputFileCreated( streamWrapper );
+                            rotationListener.outputFileCreated( bufferingOutputStream );
                         }
                     }
                     catch ( IOException e )
                     {
                         System.err.println( "Failed to open log file after log rotation: " + e.getMessage() );
-                        rotationListener.rotationError( e, streamWrapper );
+                        rotationListener.rotationError( e, bufferingOutputStream );
```

Comment:
Not a fan of that `System.err.println` – it implies a lock on stderr, an we are already capturing the error in the rotation listener.